[adot] upgrade adapter to v6

corentinverpillat opened the pull request:
Co-authored-by: Alexandre Lorin <alexandre.lorin@adotmob.com>
Co-authored-by: Corentin Verpillat <corentin.verpillat@adotmob.com>

<!--
Thank you for your pull request. Please make sure this PR is scoped to one change, and that any added or changed code includes tests with greater than 80% code coverage. See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: if checking here, also submit your bidder params documentation here https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
<!-- Describe the change proposed in this pull request -->

<!-- For new bidder adapters, please provide the following -->
- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->

aleksatr commented:
Hi @corentinverpillat, this branch has conflicts, can you rebase against master and resolve them? Also, can you describe changes you made in this PR, every single line in this adapter was changed according to the diff. Also, please bring the test coverage up to at least 80%.

corentinverpillat commented:
Hi @aleksatr, i rebased on master to fix conflicts, i fixed the tests which did not work and i added a description for my changes.
Is it ok for you ?

aleksatr commented:
> Hi @aleksatr, i rebased on master to fix conflicts, i fixed the tests which did not work and i added a description for my changes. Is it ok for you ?

@corentinverpillat "Upgrade adot adapter from v3 to v6" doesn't really tell me much, but ok :)
Anyway, you didn't bring your test coverage up to at least 80%, to check coverage:

```
gulp test-coverage
gulp view-coverage
```

Is 10s TTL for your bids intentional? This will basically mean that your bids can't be cached, in practice, when bid caching is enabled.
Did you test your adapter with price floors, I'm not that familiar with price floor module, but from what I see everyone is getting the floor from bid request, while you're calling it on ad unit `adUnit.getFloor()`, so just wanted to make sure you checked this function is actually defined on the adUnit.

corentinverpillat commented:
Hi @aleksatr, yes, 10s for the TTL is intentional.
For the module bidfloor it works like that regarding documentation and before using `adUnit.getFloor()` we check that it exists and it's a function.
And for the coverage it should be ok now, we have at least 80% on our adot adapter.
